### PR TITLE
#84 Add map directions + fix #96 and #97 + #98 Change Directions List

### LINKED ISSUE #96: The direction swap button doesn't work
Steps to Reproduce :
1. Select the search icon
1. Pick a location
1. Select the swap button

### PR DESCRIPTION
**Pull request by andrelegault**
This pull request makes it possible to view itinerary lines on the map. Additionally, it fixes bugs related to updating the UI after changing transportation modes or clicking the swap button.

To achieve this, the following changes were made:
- Add itinerary object to the state
- Update properties of the state via `changeStart, changeEnd, etc.`
- Update components via `setItinerary`

### CODE REVIEW

File: lib/models/itinerary.dart [hunk 1]
```diff
@@ -1,54 +1,97 @@
...
+    var steps = rawJson['routes'][0]['legs'][0]['steps'];
+
+    for (int i = 0; i < steps.length; i++) {
+      dynamic directions = steps[i]["polyline"]["points"];
+      List<PointLatLng> temp = tPolylinePoints.decodePolyline(directions);
+      tPointLatLng = new List.from(tPointLatLng)..addAll(temp);
+    }
+    List<LatLng> po = [];
+    tPointLatLng.forEach((f) {
+      po.add(LatLng(f.latitude, f.longitude));
```
Comment (EvanMateo7) [inline lib/models/itinerary.dart]: This is a bit hard to understand what it's doing at first glance.
Maybe add some comments for blocks of code

Comment (EvanMateo7) [reply lib/models/itinerary.dart]: Also you can remove those 4 double variables at the top if theyre not used

File: lib/widgets/map_widget.dart [hunk 2]
```diff
@@ -16,11 +16,25 @@ import 'package:concordia_navigation/models/buildingModels/building_information.
...
   @override
   _MapWidgetState createState() => _MapWidgetState();
 }
 
 class _MapWidgetState extends State<MapWidget> {
+  double _bottomSheetBottomPosition = -260;
+  bool isCollapsed = false;
+
+  _onTap() {
+    setState(() {
```
Comment (WindOfRussia) [inline lib/widgets/map_widget.dart]: Maybe we can extract this to the map data provider. 
That way we can pull up the directions list on click from a different UI component.
thoughts?

Comment (MichelRahme) [reply lib/widgets/map_widget.dart]: Fixed

File: lib/widgets/directions_drawer.dart [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+
+class DirectionsDrawer extends StatefulWidget {
+  @override
+  _DirectionsDrawerState createState() => _DirectionsDrawerState();
+}
+
+class _DirectionsDrawerState extends State<DirectionsDrawer> {
+  bool isCollapsed = false;
```
Comment (WindOfRussia) [inline lib/widgets/directions_drawer.dart]: I think you forgot to remove this. Might as well make this a stateless widget.

Comment (MichelRahme) [reply lib/widgets/directions_drawer.dart]: Yes, you're right, going to convert it to Stateless Widget.

File: lib/models/itinerary.dart (hunk 1 above)
Comment (andrelegault) [reply lib/models/itinerary.dart]: I'm gonna keep the ones for involving starting location info i.e. `sLat, sLong, and sDesc` for now. They might be used in when viewing the list of instructions in (`/lib/widgets/directions_drawer.dart`)

File: lib/providers/map_data.dart [hunk 4]
```diff
@@ -1,17 +1,23 @@
...
 class MapData extends ChangeNotifier {
   Completer<GoogleMapController> _completer = Completer();
+  Itinerary itinerary;
 
   Completer<GoogleMapController> get getCompleter {
     return _completer;
   }
 
+  final double _expandedBottomSheetBottomPosition = 0;
+  final double _collapsedBottomSheetBottomPosition = -700;
```
Comment (WindOfRussia) [inline lib/providers/map_data.dart]: Perhaps we can use `SizeConfig` or move these contants to some other place?

Comment (MichelRahme) [reply lib/providers/map_data.dart]: Oh yeah, completely forgot to use the `SizeConfig()` class

File: lib/models/itinerary.dart (hunk 1 above)
Comment (andrelegault) [reply lib/models/itinerary.dart]: fixed in 29ee4c271ab5830164da28b1f0d65a239c921d08

**Pull request merged**